src/main/java/seedu/address/logic/parser/FitAddCommandParser.java

```diff
@@ -0,0 +1,132 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EXERCISE_ABS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EXERCISE_ALL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EXERCISE_ARM;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EXERCISE_BACK;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EXERCISE_BREAK_BETWEEN_SETS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EXERCISE_CHEST;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EXERCISE_LEG;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EXERCISE_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EXERCISE_REPS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EXERCISE_SETS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EXERCISE_SHOULDER;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.FitAddCommand;
+import seedu.address.logic.messages.FitAddCommandMessages;
+import seedu.address.logic.messages.Messages;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.exercise.Exercise;
+
+/**
+ * Parses input arguments and creates a new FitAddCommand object
+ */
+public class FitAddCommandParser implements Parser<FitAddCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FitAddCommand
+     * and returns a FitAddCommand object for execution.
+     *
+     * @param args The string of arguments to be parsed
+     * @return The FitAddCommand object for execution
+     * @throws ParseException If the user input does not conform to the expected format
+     */
+    public FitAddCommand parse(String args) throws ParseException {
```

Comment:
You have a few methods that are significantly longer than the [recommened length](https://nus-cs2103-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods) of 30 LoC.

Maybe they could be broken up to improve code quality?